### PR TITLE
[ESP32] ESP-IDF v5.0 support in few apps

### PR DESCRIPTION
**Pull request by shubhamdp**
#### Problem
- Related to #24576

#### Change Overview
Added support for esp-idf v5.0 in following applications all-clusters-app, all-clusters-minimal-app, bridge-app, and light-switch-app

In a nutshell:
- Update m5stack-tft submodule to support idf v5.0
- Fixed few format specifier
- Changed `ErrorStr` to `err.Format()`
- Added components to required list. In v4.4.4 these made it to list somehow through some other component's global requirement list but, those are not made private and hence needed to be added explicitly.
- Few APIs are changed in v5.0, so handling them using ifdefs.

#### Testing
- Successfully compiled above apps for esp32 and esp32c3 on idf v5.0 and v4.4.3

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -74,3 +74,7 @@ examples/thermostat/ameba/build
 
 # Downloaded zap without a pigweed root (via zap_download.py)
 .zap
+
+# for idf component manager files
+examples/*/esp32/managed_components
```
Comment (andy31415) [inline .gitignore:79]: Why do we expect these to appear? Could we create an out-of-tree build that does not dirty the source tree?

**Review by andy31415 — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (shubhamdp) [reply .gitignore:79]: There's been a lot of restructuring in esp-idf v5.0, most of non-core component are being moved out to [component registry](https://components.espressif.com/). These components are fetched using [idf-component-manager](https://github.com/espressif/idf-component-manager) and stored at `project/managed_components` making the tree dirty.

I'll check if there's a way to build examples out-of-tree and if I found something I will put up a follow up PR?

Comment (andy31415) [reply .gitignore:79]: Could we provide a definitive answer (either find a workaround or exaplain why a workaround is not possible in the comment here)? I am concerned that "if something will be found, a future PR will follow" is likely to result in this not being done.

At least if followup PR seems to be needed (you cannot find something within a reasonable time), please document here some explanation and links about managed_components and why this exists.

Comment (shubhamdp) [reply .gitignore:81]: I have added a comment and a link stating why those files are generated

**Review by chshu — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**